### PR TITLE
Improve I18N Issues based on 4.1

### PR DESCRIPTION
**Pull request by alexclassroom**
- Improve I18N issues
- Change some code. In the original code, if the number is over 1, "minute" and "hour" will be added **s** at the end, but not all languages are like this. If "minute" and "hour" are translated, and the number is over 1, for example, zh_TW, the UI will show "2 小時**s**" or "2 分鐘**s**". This is so weird.

### CODE REVIEW

**Comment by alexclassroom** [comment]
If your other plugins used similar **About** page content, it is a good idea to fix them and improve the same I18N issues.

File: admin/partials/jws-about.php [hunk 1]
```diff
@@ -20,22 +20,22 @@
         <img src="<?php echo esc_attr( plugins_url( '../images/icon-256x256.png', __FILE__ ) ); ?>">
 
         <h2><?php printf( esc_html__( 'Just Writing Statistics V%s', 'just-writing-statistics' ), JWS_VERSION ); ?></h2>
-        <p><?php esc_html_e( 'a fork of', ''); ?> <a href="https://wpwordcount.com">WP Word Count</a></p>
-        <p><?php esc_html_e( 'by', ''); ?> <a href="https://toolstack.com">Greg Ross</a></p>
+        <p><?php printf(esc_html__('a fork of %1$sWP Word Count%2$s', 'just-writing-statistics'),'<a href="https://wpwordcount.com" target="_blank">','</a>'); ?>
```
Comment (toolstack) [inline admin/partials/jws-about.php:23]: "WP Word Count" shouldn't be part of the translation string, I'd be fine with something like:

```
printf(esc_html__('a fork of %1$s', 'just-writing-statistics'),'<a href="https://wpwordcount.com" target="_blank">WP Word Count</a>');
```

File: admin/partials/jws-about.php [hunk 2]
```diff
@@ -20,22 +20,22 @@
...
+        <p><?php printf(esc_html__('by %1$sGreg Ross%2$s', 'just-writing-statistics'),'<a href="https://toolstack.com" target="_blank">','</a>'); ?></p>
```
Comment (toolstack) [inline admin/partials/jws-about.php:24]: Same as above.

File: includes/class-jws-functions.php [hunk 3]
```diff
@@ -154,15 +154,15 @@ function jws_reading_time($word_count, $wpm = 250, $format = 'admin')
 
     if ($format == 'admin') {
         if ($minutes == 0) {
-            $html = '<1 minute';
+            $html = __( '<1 minute', 'just-writing-statistics' );
         } else {
             if ($hours == 0) {
-                $html = number_format($minutes).' minute'.($minutes > 1 ? 's' : '');
+                $html = number_format($minutes).__( ' minute(s)', 'just-writing-statistics' );
```
Comment (toolstack) [inline includes/class-jws-functions.php:160]: Agree the if statement for 's' is incorrect, but the fix isn't '(s)', but instead to do two separate strings, one 'minute' and one 'minutes'.

File: includes/class-jws-functions.php [hunk 4]
```diff
@@ -154,15 +154,15 @@ function jws_reading_time($word_count, $wpm = 250, $format = 'admin')
...
             } else
             if ($minutes == 0) {
-                $html = number_format($hours).' hour'.($hours > 1 ? 's' : '');
+                $html = number_format($hours).__( ' hour(s)', 'just-writing-statistics' );
```
Comment (toolstack) [inline includes/class-jws-functions.php:163]: Same as above.

File: includes/class-jws-functions.php [hunk 5]
```diff
@@ -154,15 +154,15 @@ function jws_reading_time($word_count, $wpm = 250, $format = 'admin')
...
             } else {
-                $html = number_format($hours).' hour'.($hours > 1 ? 's' : '').', '.number_format($minutes).' minute'.($minutes > 1 ? 's' : '');
+                $html = number_format($hours).__( ' hour(s)', 'just-writing-statistics' ).', '.number_format($minutes).__( ' minute(s)', 'just-writing-statistics' );
```
Comment (toolstack) [inline includes/class-jws-functions.php:165]: Same as above.

**Review by toolstack — COMMENTED** [review]
Thanks for the PR, see the comments for a few minor changes.

File: admin/partials/jws-about.php (hunk 1 above)
Comment (alexclassroom) [reply admin/partials/jws-about.php:23]: Keeping complete sentences is a great choice, so just adding a comment to let the translators know not to translate the plugin name "WP Word Count".

File: admin/partials/jws-about.php (hunk 2 above)
Comment (alexclassroom) [reply admin/partials/jws-about.php:24]: Keeping complete sentences is a great choice, so just adding a comment to let the translators know not to translate the developer name "Greg Ross".

File: includes/class-jws-functions.php (hunk 3 above)
Comment (alexclassroom) [reply includes/class-jws-functions.php:160]: Agree.

File: includes/class-jws-functions.php (hunk 4 above)
Comment (alexclassroom) [reply includes/class-jws-functions.php:163]: Agree.

**Comment by toolstack** [comment]
Closing this PR as I committed a slightly different version of the changes.

Thanks @alexclassroom for submitting this.

**Pull request closed without merging**